### PR TITLE
Adding the Support to Proxy the call over to the _agent for Actions API

### PR DESCRIPTION
**Pull request by aditjind**
*Fixes #, if available:* Adding the Support to Proxy the call over to the _agent for Actions API

*Description of changes:* Add the proxy support as well as enable tests which were not enabled due to `Test` file name instead of a `Tests` filename.  

*Tests:*

=======================================

> Task :spotbugsMain
Warning at xsl:variable on line 348 column 57 of default.xsl:
  SXWN9001: A variable with no following sibling instructions has no effect
Warning at xsl:variable on line 351 column 57 of default.xsl:
  SXWN9001: A variable with no following sibling instructions has no effect

Deprecated Gradle features were used in this build, making it incompatible with Gradle 7.0.
Use '--warning-mode all' to show the individual deprecation warnings.
See https://docs.gradle.org/6.0.1/userguide/command_line_interface.html#sec:command_line_warnings

BUILD SUCCESSFUL in 14s

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: src/test/java/com/amazon/opendistro/elasticsearch/performanceanalyzer/http_action/config/PerformanceAnalyzerResourceProviderTests.java [hunk 1]
```diff
@@ -43,7 +43,7 @@
 @RunWith(PowerMockRunner.class)
 @PrepareForTest({PluginSettings.class})
 @SuppressStaticInitializationFor({"PluginSettings"})
-public class PerformanceAnalyzerResourceProviderTest {
+public class PerformanceAnalyzerResourceProviderTests {
```
Comment (vigyasharma) [inline src/test/java/com/amazon/opendistro/elasticsearch/performanceanalyzer/http_action/config/PerformanceAnalyzerResourceProviderTests.java:46]: Why was this rename needed?

**Review by vigyasharma — APPROVED** [review]

File: src/test/java/com/amazon/opendistro/elasticsearch/performanceanalyzer/http_action/config/PerformanceAnalyzerResourceProviderTests.java (hunk 1 above)
Comment (aditjind) [reply src/test/java/com/amazon/opendistro/elasticsearch/performanceanalyzer/http_action/config/PerformanceAnalyzerResourceProviderTests.java:46]: To run these UTs while building. In out gradle setup we filter test files via `*Tests` Suffix. These were not running till now.

File: src/test/java/com/amazon/opendistro/elasticsearch/performanceanalyzer/http_action/config/PerformanceAnalyzerResourceProviderTests.java [hunk 2]
```diff
@@ -133,6 +133,17 @@ private void assertAgentUriWithBatchRedirection(final String protocolScheme) thr
     assertEquals(new URL(expectedResponseUri), actualResponseURI);
   }
 
+  private void assertAgentUriWithActionsRedirection(final String protocolScheme) throws IOException {
+    initPerformanceAnalyzerResourceProvider(protocolScheme.equals("https://"));
+
+    String requestUri = protocolScheme + "localhost:9200/_opendistro/_performanceanalyzer/_agent/actions";
+    String expectedResponseUri = protocolScheme + "localhost:9650/_opendistro/_performanceanalyzer/actions";
```
Comment (yojs) [inline src/test/java/com/amazon/opendistro/elasticsearch/performanceanalyzer/http_action/config/PerformanceAnalyzerResourceProviderTests.java:140]: Instead of hardcoding, should we pick it from `Util::RPC_PORT` ?

**Review by yojs — APPROVED** [review]
